### PR TITLE
Add zero copy version of pre-process json decoder

### PR DESCRIPTION
**Pull request by NeatGuyCoding**
Result:
```

Benchmark                                    Mode  Cnt      Score     Error   Units
PreprocessJsonBenchmark.benchmarkNewMethod  thrpt    5  19135.213 ± 150.611  ops/ms
PreprocessJsonBenchmark.benchmarkOldMethod  thrpt    5   8616.179 ± 146.673  ops/ms
```

## Summary by CodeRabbit

* **Chores**
  * Updated Maven build pipeline configuration to run enhanced validation steps during builds.
  * Added code quality checks across multiple project modules.

* **Tests**
  * Introduced performance benchmarking tests to validate system efficiency.
  * Expanded test coverage with comprehensive scenarios and edge cases.

### CODE REVIEW

File: netty-socketio-core/src/main/java/com/corundumstudio/socketio/protocol/PacketDecoder.java [hunk 1]
```diff
@@ -50,24 +49,143 @@ private boolean isStringPacket(ByteBuf content) {
...
-            *
-            * @see https://github.com/Automattic/socket.io-client/blob/1.3.3/socket.io.js#L2682
-            */
-            packet = packet.replace("\\\\n", "\\n");
-
-            // skip "d="
-            packet = packet.substring(2);
+        // Create a mutable copy of the input ByteBuf for in-place modifications
+        ByteBuf mutableContent = content.copy();
+
```
[the remark this answers is not in the document]
Comment (NeatGuyCoding) [reply netty-socketio-core/src/main/java/com/corundumstudio/socketio/protocol/PacketDecoder.java]: thanks, I think I will use slice because the original caller seems not release the buffer
